ui/dashboards/src/components/PanelDrawer/PanelDrawer.test.tsx

```diff
@@ -0,0 +1,102 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { PluginRegistry } from '@perses-dev/plugin-system';
+import { screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import * as dashboardAppSlice from '../../context/DashboardAppSlice';
+import * as layoutsSlice from '../../context/LayoutsSlice';
+import * as context from '../../context/DashboardProvider';
+import { FAKE_PANEL_PLUGIN, mockPluginRegistryProps, renderWithContext } from '../../test';
+import testDashboard from '../../test/testDashboard';
+import PanelDrawer from './PanelDrawer';
+
+const updatePanel = jest.fn();
+jest.spyOn(context, 'usePanels').mockReturnValue({
+  updatePanel,
+  panels: {},
+});
+
+const addItemToLayout = jest.fn();
+jest.spyOn(layoutsSlice, 'useLayouts').mockReturnValue({
+  addItemToLayout,
+  updateLayout: jest.fn(),
+  layouts: testDashboard.spec.layouts,
+});
```

Comment:
Just a general observation: we've largely tried to avoid mocking hooks (with the idea that we get more value out of testing the whole flow with the "real" logic). Any way to test this stuff without mocking them? Fine if not, just wanted to ask.

Reply:
Side note: if you're doing this because you want to write more "unit" style tests, that's totally cool too. Don't think we've really embraced a "testing philosophy" here in Perses the way we have with some of our internal code where we tend to favor "integration" style tests, so feel free to go for it if that's what you like.

Reply:
Yep this is meant to be a unit test, thinking we'll add integration tests later

Also for integration tests, i think i'll have to first figure out how to mock the zustand store